src/main/scala/com/vngrs/etl/Pipeline.scala

```diff
@@ -0,0 +1,97 @@
+
+  def load(loaders: Loader[A]*): Unit = loaders.foreach(loader => loader(rdd))
+
+  def zip[B: ClassTag](other: Pipeline[B]): Pipeline[(A, B)] = new Pipeline(rdd.zip(other.rdd))
+
+  def union(other: Pipeline[A]): Pipeline[A] = new Pipeline(rdd.union(other.rdd))
+}
+
+object Pipeline {
+  def apply[A: ClassTag](extractor: Extractor[A])(implicit sc: SparkContext): Pipeline[A] = Pipeline(extractor(sc))
+
+  def apply[A: ClassTag](iterable: Iterable[A])(implicit sc: SparkContext): Pipeline[A] = Pipeline(sc.parallelize[A](iterable.toSeq))
+
+  def apply[A: ClassTag](arr: Array[A])(implicit sc: SparkContext): Pipeline[A] = Pipeline(sc.parallelize[A](arr))
+
+  def apply[A: ClassTag](rdd: RDD[A]): Pipeline[A] = new Pipeline(rdd)
+}
+
+private object UsageExamples {
+
+  implicit val sc: SparkContext = new SparkContext()
+
+  val extractor = new Extractor[String] {
+    override def apply(sc: SparkContext): RDD[String] = sc.parallelize[String](Seq("A", "B", "C"))
+  }
+
+  val transformer = new Transformer[String, String] {
+    override def apply(data: RDD[String]): RDD[String] = data.map(_.toLowerCase())
+  }
+
+  val squareTrans = new Transformer[Int, Double] {
+    override def apply(data: RDD[Int]): RDD[Double] = data.map(Math.pow(2, _))
+  }
+
+  val loader = new Loader[String] {
+    override def apply(data: RDD[String]): Unit = data.saveAsTextFile("path")
+  }
+
+  val extractorNum = new Extractor[Int] {
+    override def apply(sc: SparkContext): RDD[Int] = sc.parallelize[Int](Seq(1, 2, 3))
+  }
+
+  val transformerNum = new Transformer[(String, Int), String] {
+    override def apply(data: RDD[(String, Int)]): RDD[String] = data.map(t => s"${t._1}|${t._2}")
+  }
+
+  val loader2 = new Loader[String] {
+    override def apply(data: RDD[String]): Unit = data.saveAsTextFile("path")
+  }
+
+  final case class PowTrans(pow: Double) extends Transformer[Int, Double] {
+    override def apply(data: RDD[Int]): RDD[Double] = data.map(Math.pow(pow, _))
+  }
+
+  final case class FileLoader(path: String) extends Loader[String] {
+    override def apply(data: RDD[String]): Unit = data.saveAsTextFile(path)
+  }
+
+  Pipeline(extractor)
+    .transform(transformer)
+    .load(loader)
+
+  Pipeline(extractor)
+    .zip(Pipeline(extractorNum))
+    .transform(transformerNum)
+    .load(loader)
+
+  Pipeline(extractor)
+    .zip(Pipeline(extractorNum))
+    .transform(transformerNum)
+    .load(loader, loader2)
+
+  Pipeline(extractorNum)
+    .transform(PowTrans(2.0d))
+    .transform(Transformer.filter(_ > 15))
+    .zip(
+      Pipeline(extractorNum)
+        .transform(squareTrans)
+        .transform(Transformer.filter(_ > 3))
+    )
+    .transform(Transformer(_.toString))
+    .load(FileLoader("path/to/path"))
+
+  val pipeline1 = Pipeline(extractorNum)
+  val pipeline2 = Pipeline(extractorNum)
+
+  val zippedPipeline = Aggregator.zip(pipeline1, pipeline2)
```

Comment:
Ok, why don't we implement some `combine` methods inside `Pipeline`?

```scala
sealed trait Strategy
case object Zip extend Strategy
case object Union extends Strategy

class Pipeline(rdd: RDD) {
  //for combining with another pipeline
  def combine(other: Pipeline, strategy: Strategy): Pipeline = strategy match {
    case Zip => rdd.zip(other.rdd)
    case Union => rdd.union(other.rdd)
  }

  //for combining with another rdd(I don't know if it's correct for this to be here)
  def combine(other: RDD) = ...
}
```

And if these strategies should be extendable as well, we can create an apply method within them, in the end they'll all combine two RDDs.
```scala
trait Strategy
trait Zip extends Strategy {
  def apply(first: RDD, second: RDD) = first.zip(second)
}
```
And we can also combine unlimited numbers of RDDs. Here's a code snippet from akka streams for combining sources source 
```scala
 /**
   * Combines several sources with fun-in strategy like `Merge` or `Concat` and returns `Source`.
   */
  def combine[T, U](first: Source[T, _], second: Source[T, _], rest: Source[T, _]*)(strategy: Int ⇒ Graph[UniformFanInShape[T, U], NotUsed]): Source[U, NotUsed] =
    Source.fromGraph(GraphDSL.create() { implicit b ⇒
      import GraphDSL.Implicits._
      val c = b.add(strategy(rest.size + 2))
      first ~> c.in(0)
      second ~> c.in(1)

      @tailrec def combineRest(idx: Int, i: Iterator[Source[T, _]]): SourceShape[U] =
        if (i.hasNext) {
          i.next() ~> c.in(idx)
          combineRest(idx + 1, i)
        } else SourceShape(c.out)

      combineRest(2, rest.iterator)
    })
```